pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -132,23 +132,27 @@ func (recycler *PersistentVolumeRecycler) handleRecycle(pv *api.PersistentVolume
 	spec := volume.NewSpecFromPersistentVolume(pv, false)
 	plugin, err := recycler.pluginMgr.FindRecyclablePluginBySpec(spec)
 	if err != nil {
-		return fmt.Errorf("Could not find recyclable volume plugin for spec: %+v", err)
-	}
-	volRecycler, err := plugin.NewRecycler(spec)
-	if err != nil {
-		return fmt.Errorf("Could not obtain Recycler for spec: %+v", err)
-	}
-	// blocks until completion
-	err = volRecycler.Recycle()
-	if err != nil {
-		glog.Errorf("PersistentVolume[%s] failed recycling: %+v", pv.Name, err)
-		pv.Status.Message = fmt.Sprintf("Recycling error: %s", err)
 		nextPhase = api.VolumeFailed
-	} else {
-		glog.V(5).Infof("PersistentVolume[%s] successfully recycled\n", pv.Name)
-		nextPhase = api.VolumePending
+		pv.Status.Message = fmt.Sprintf("%v", err)
+	}
+
+	if plugin != nil {
```

Comment:
I can't do that here (same with your comment in the handleDelete function) because the phase is set to Failed in the code above and must be persisted later in the function.

I added a comment to both that explains what's going on and why to make it clear.